### PR TITLE
Results error handling

### PR DESCRIPTION
**Pull request by bmckinney-ebsco**
- addresses #90 and #88
- VCR tests and changelog updated

### CODE REVIEW

File: lib/ebsco/eds/results.rb [hunk 1]
```diff
@@ -32,73 +32,81 @@ class Results
...
-            #   else
-            #     @records.push(EBSCO::EDS::Record.new(record))
-            #   end
-            # else
-            #   @records.push(EBSCO::EDS::Record.new(record))
-            # end
 
-          }
-        end
+        if Hash === @results
```
Comment (jkeck) [inline lib/ebsco/eds/results.rb]: A more idiomatic approach to this would be to check `if @results.is_a?(Hash)`.

File: lib/ebsco/eds/results.rb [hunk 2]
```diff
@@ -32,73 +32,81 @@ class Results
...
-        end
 
-        # titleize facets
-        @titleize_facets = %w[Language Journal SubjectEDS SubjectGeographic Publisher]
+          # titleize facets
+          @titleize_facets = %w[Language Journal SubjectEDS SubjectGeographic Publisher]
+
+        else
+          # response isn't a hash (eg, html error page)
+          raise EBSCO::EDS::ApiError, 'EBSCO API error: Query failed.'
```
Comment (jkeck) [inline lib/ebsco/eds/results.rb:108]: Just to clarify, is the expectation that downstream applications will capture for this and display whatever messaging to their end users?

**Review by jkeck — COMMENTED** [review]
Just a few minor comments.  Would you like me to try this out in SearchWorks as well?

File: lib/ebsco/eds/results.rb (hunk 2 above)
Comment (bmckinney-ebsco) [reply lib/ebsco/eds/results.rb:108]: That's right. But open to suggestions for a better approach.

File: lib/ebsco/eds/results.rb (hunk 1 above)
Comment (bmckinney-ebsco) [reply lib/ebsco/eds/results.rb]: Ah, thanks! Will modify this.

**Comment by bmckinney-ebsco** [comment]
@jkeck I just committed some dependency updates to this branch, would appreciate SearchWorks testing when time permits. Thanks!

**Comment by jkeck** [comment]
Just a heads up @bmckinney, I'm off for a few days.  Will try to take this for a spin in SearchWorks next week.

**Pull request merged**